pkg/model/hornet/message_id.go

```diff
@@ -0,0 +1,148 @@
+func (m MessageID) ToMapKey() string {
+	return string(m)
+}
+
+// GetNullMessageID returns the ID of the genesis message.
+func GetNullMessageID() MessageID {
+	nullMessageID := make(MessageID, 32)
+	return nullMessageID
+}
+
+// MessageIDFromHex creates a MessageID from a hex string representation.
+func MessageIDFromHex(hexString string) (MessageID, error) {
+
+	b, err := hex.DecodeString(hexString)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(b) != iotago.MessageIDLength {
+		return nil, fmt.Errorf("unknown messageID length (%d)", len(b))
+	}
+
+	return MessageID(b), nil
+}
+
+// MessageIDFromMapKey creates a MessageID from a map key representation.
+func MessageIDFromMapKey(mapKey string) MessageID {
+	if len(mapKey) != iotago.MessageIDLength {
+		panic(fmt.Sprintf("unknown messageID length (%d)", len(mapKey)))
+	}
+
+	return MessageID(mapKey)
+}
+
+// MessageIDFromSlice creates a MessageID from a byte slice.
+func MessageIDFromSlice(b []byte) MessageID {
+
+	if len(b) != iotago.MessageIDLength {
+		panic(fmt.Sprintf("unknown messageID length (%d)", len(b)))
+	}
+
+	return MessageID(b)
+}
+
+// MessageIDFromArray creates a MessageID from a byte array.
+func MessageIDFromArray(b iotago.MessageID) MessageID {
+	return MessageID(b[:])
+}
+
+// ToHex converts the MessageIDs to their hex string representation.
+func (m MessageIDs) ToHex() []string {
+	var results []string
+	for _, msgID := range m {
+		results = append(results, msgID.ToHex())
+	}
+	return results
+}
+
+// ToSliceOfSlices converts the MessageIDs to a slice of byte slices.
+func (m MessageIDs) ToSliceOfSlices() [][]byte {
+	var results [][]byte
+	for _, msgID := range m {
+		// this is necessary, otherwise we create a pointer to the loop variable
+		tmp := msgID
+		results = append(results, tmp)
+	}
+	return results
+}
+
+// ToSliceOfArrays converts the MessageIDs to a slice of byte arrays.
+func (m MessageIDs) ToSliceOfArrays() iotago.MessageIDs {
+	var results iotago.MessageIDs
+	for _, msgID := range m {
+		results = append(results, msgID.ToArray())
+	}
+	return results
+}
+
+// RemoveDupsAndSortByLexicalOrder returns a new slice of MessageIDs sorted by lexical order and without duplicates.
+func (m MessageIDs) RemoveDupsAndSortByLexicalOrder() MessageIDs {
+
+	seen := make(map[string]struct{})
+	orderedArray := make(iotago.LexicalOrderedByteSlices, len(m))
+
+	uniqueElements := 0
+	for _, v := range m {
+		// this is necessary, otherwise we create a pointer to the loop variable
+		tmp := v
+		k := string(v)
+		if _, has := seen[k]; has {
+			continue
+		}
+		seen[k] = struct{}{}
+		orderedArray[uniqueElements] = tmp
+		uniqueElements++
+	}
+	orderedArray = orderedArray[:uniqueElements]
+	sort.Sort(orderedArray)
+
+	result := make(MessageIDs, uniqueElements)
+	for i, v := range orderedArray {
+		// this is necessary, otherwise we create a pointer to the loop variable
+		tmp := v
+		result[i] = tmp
+	}
+
+	return result
```

Comment:
```suggestion
	// sort the messages lexicographically
	sorted := make(iotago.LexicalOrderedByteSlices, len(m))
	for i, id := range m {
		sorted[i] = id
	}
	sort.Sort(sorted)

	var result MessageIDs
	var prev []byte
	for i, id := range sorted {
		// only add to the result, if it its different from its predecessor
		if i == 0 || !bytes.Equal(prev, id) {
			result = append(result, id)
		}
		prev = id
	}
	return result
```
This would be an alternative implementation. That could be much cleaner.
It might be a bit slower for huge inputs with many duplicates (as sorting is n*log(n) while map accesses - excluding resizing - are amortized n), while it is faster for inputs with few duplicates. If that is of any consideration, the current solution is fine, although `tmp` is unnecessary.